src/liquid/place/place_navbar.html

```diff
@@ -0,0 +1,25 @@
+<div class="topnav" id="topnav">
+    {% for contactType in contactTypes %}
+    {% capture tabClass %}
+    {% if contactType.name == directiveModel.activeTab %}active
+    {%endif%}
+    {%endcapture%}
+    <a hx-trigger="click" hx-get="/app/list?page=1&contactTypeName={{ contactType.name }}"
+        id="tab_{{contactType.name}}" hx-target="#place_list" hx-swap="outerHTML show:window:top" onclick="highlight(this)"
+        class="{{tabClass}} nav-item">{{contactType.friendly}}
+        {% include "components/nav_badge.html" %}
+    </a>
+    {% endfor %}
+    <a hx-trigger="click" hx-get="/app/list/summary"
+    {% capture summarytabClass %} {% if directiveModel.activeTab == 'summary' %}active{%endif%}{%endcapture%}
+    id="tab_summary" hx-target="#place_list" hx-swap="outerHTML show:window:top" class="nav-item {{summarytabClass}}" onclick="highlight(this)">Summary</a>
+</div>
+
+<script>
+    function highlight(element) {
```

Comment:
Anything ui related that can be done frontend shouldnt do around trip to the server,  find it unnecessary

Reply:
While i agree with you, you still make the trip to the server to fetch the page after clicking the tab. Being a server rendered app, the server manages most of the state in the UI. In this case, you can return both the page and the updated navbar in the response from the server and achieve the same effect without JS

Reply:
but htmx, sse etc all run in the frontend and do manipulate ui before even hitting the server!
this is equivalent to asking the server to register/listen for button clicks 
i dont think thats what SSR is

Reply:
> but htmx, sse etc all run in the frontend and do manipulate ui before even hitting the server!
this is equivalent to asking the server to register/listen for button clicks

There's nothing fancy going on. We're just issuing AJAX calls to the server on certain events. htmx helps with this and content loading/swapping. We can ditch htmx and just use full page reloads for any state change and the app would work the same as the state is all on the server.

Reply:
ditching it would be pointless. you can never get rid of js completely  in web development even if your back end is python
JS was built for web development
The point i was making wasnt about htmx only but all the js served in the public folder all of it runs in the browser 
all the calendar js tooltip js etc

Reply:
Yes i understand some UI elements are complex and need JS to manage their state. To be clear, I'm not saying we should get rid of JS, I'm saying that the application state should be driven by the server as much as possible. Application state in this app should be the same with or without JS

Reply:
is this why we have things like this [place.className](https://github.com/medic/cht-user-management/blob/136998e2c79015d08cbb80f9ad4c6fcfab80edf3/src/liquid/components/button_upload.html#L1)? server literally storing and setting classes? you dont find this weird?
UI state and server state should be managed seperately, with only communications between the 2 on the state of  one  another

Reply:
All I'm saying is update the tab's state by the same way the contents of the page are updated. We don't need a script to do that.

Reply:
this has been changed